src/common/numa.cc

```diff
@@ -156,6 +157,24 @@ static int easy_readdir(const std::string& dir, std::set<std::string> *out)
   return 0;
 }
 
+std::string get_thread_name(char *thread_file)
```

Comment:
- i'd avoid using the same name used in `src/common/Thread.h`.
- and should handle `EINTR` when performing syscall
- replace the trailing newline at the right position
- add space after comma
- remove space in `read (`.
- indent properly

so i propose:
```c++
std::string get_task_comm(pid_t tid)
{
  static const char* comm_fmt = "/proc/self/task/%d/comm";
  char comm_name[sizeof(comm_fmt) + 8];
  snprintf(comm_name, sizeof(comm_name), comm_fmt, tid);
  int fd = open(comm_name, O_CLOEXEC | O_RDONLY);
  if (fd == -1) {
    return "";
  }
  // see linux/sched.h
  static constexpr int TASK_COMM_LEN = 16;
  char name[TASK_COMM_LEN];
  ssize_t n = safe_read(fd, name, sizeof(name));
  close(fd);
  if (n < 0) {
    return "";
  }
  assert(n < sizeof(name));
  if (name[n - 1] == '\n')
    name[n - 1] = '\0';
  } else {
    name[n] = '\0';
  }
  return name;
}
```